content/Contributing to open source: An overview of the django-drip-campaings project - Bruno Michetti.md

```diff
@@ -0,0 +1,217 @@
+  |__ docs/    # Documentation files
+  |__ project_folder/  # Code of the app
+  |__ AUTHORS.md
+  |__ LICENSE.txt
+  |__ MANIFEST.in
+  |__ README.md
+  |__ setup.py
+```
+
+- AUTHORS: Specify the author's information of your project in this file.
+
+- LICENSE: You have to specify the license of the project. That is a document that provides legally binding guidelines for the use and distribution of software. In OSS the most known are [BSD](https://en.wikipedia.org/wiki/BSD_licenses) and [MIT](https://en.wikipedia.org/wiki/MIT_License). We used an MIT license in this case.
+
+- MANIFEST: Here you can specify the files that aren't programming files but you want to include anyway in the build of your project. For example the authors file, the docs, etc.
+
+- README.md: A file where you introduce and explain your project. It's very important because otherwise you can have a very good and useful project but if new people see it and don't understand what it does nor how it works, then it won't be used.
+
+- setup.py: This is a fundamental file. It contains a global `setup` function that is in charge of the building and configuration of your project at the moment of the installation.
+
+- docs folder: Here is where the files of the documentation are.
+
+You can see examples of the described files in the [django-drip-campaigns github page](https://github.com/rootstrap/django-drip-campaigns). Of course your project can have other files, the ones mentioned are the most important to publish a pypi package.  
+Having all set and ready to be published, you need now to [register in the pypi.org page](https://pypi.org/account/register/). After you are registered, go inside the terminal to your project folder and follow these steps:
+
+1. Install `setuptools` and `wheel`. They are used to generate the distribution package: a group of files needed to upload your app.
+
+```
+python3 -m pip install --user --upgrade setuptools wheel
+```
+
+2. Generate the distribution package. The next command will create all the necessary files under a `dist/` folder:
+
+```
+python3 setup.py sdist bdist_wheel
+```
+
+3. Install [twine](https://packaging.python.org/key_projects/#twine), that a allows you to upload your app as a package:
+
+```
+python3 -m pip install --user --upgrade twine
+```
+
+4. Once installed, run twine to upload the archives under `dist/`. This will ask for a user and a password, that are the ones you used on the registration at pypi:
+
+```
+python3 -m twine upload --repository pypi dist/*
+```
+
+If everything went well, you have published your project as a pypi package. Now you anyone is able to download it with:
+
+```
+pip3 install <project-name>
+```
+
+### Summary
+
+In this blog I talked about the Django Drip Campaigns project, the main features and what we did with it at Rootstrap. We love to make contributions to OSS because it's a wonderful way of learning, of sharing knowledge, and creating useful programs to be used by other programmers.  
```

Comment:
```suggestion
In this blog post I talked about the Django Drip Campaigns project, its main features and what we did with it at Rootstrap. We love to make contributions to OSS because it's a wonderful way of learning, of sharing knowledge, and creating useful programs to be used by other programmers.  
```